### PR TITLE
Properly handle Mandatory visual elements' mandatory properties.

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Addresses issue #110.

Also, to change the order of properties, one can simply create a property entry (potentially with only the name). Inherited doc (type, default, description) will be copied to empty field.

### CODE REVIEW

File: docs/manuals/gui/pages.md [hunk 1]
```diff
@@ -84,7 +84,7 @@ whose content is defined by Markdown text.
     Markdown fragment in the definition of "page1":
     ```
     ...
-    Goto to [Second Page](/page2) for more information.
+    Go to [Second Page](/page2) for more information.
```
Comment (FredLL-Avaiga) [inline docs/manuals/gui/pages.md:87]: good catch

File: docs/manuals/gui/pages.md [hunk 2]
```diff
@@ -257,7 +257,7 @@ root page of your application, and is replaced by the target page content when t
        pages = {
          "/": root_md,
          "page1": page1_md,
-         "page1": page2_md
+         "page2": page2_md
```
Comment (FredLL-Avaiga) [inline docs/manuals/gui/pages.md:260]: 👍

File: tools/setup_generation.py [hunk 3]
```diff
@@ -130,7 +130,14 @@ def read_skeleton(name):
 # -----------------------------------------------------------------------------
 element_properties = {}
 element_documentation = {}
-
+INHERITED_PROP = 1<<0
```
Comment (FredLL-Avaiga) [inline tools/setup_generation.py:133]: the C expert in you

File: tools/setup_generation.py [hunk 4]
```diff
@@ -140,63 +147,72 @@ def read_properties(path_name, element_name):
...
+                        # Testing for equality detects NaN
+                        properties[prop_index] = (p[0], p[1],
+                                                  p[2] if p[2] == p[2] else parent_prop[2],
+                                                  p[3] if p[3] == p[3] else parent_prop[3],
+                                                  p[4] if p[4] == p[4] else parent_prop[4])
+                    except:
+                        properties.append(parent_prop)
             else:
+                row[0] = prop_name
+                row = (prop_flags, prop_name, *row.tolist()[1:])
```
Comment (FredLL-Avaiga) [inline tools/setup_generation.py:181]: so that's how you create a variable size tuple ...

**Review by FredLL-Avaiga — APPROVED** [review]
🥇 
i'd say cleaner and easier to evolve with prop_flags

File: tools/setup_generation.py (hunk 3 above)
Comment (FabienLelaquais) [reply tools/setup_generation.py:133]: 😒

**Pull request merged**